test/module/irohad/ametsuchi/postgres_options_test.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2018 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ametsuchi/impl/postgres_options.hpp"
+#include <gtest/gtest.h>
+#include <boost/algorithm/string.hpp>
+
+using namespace iroha::ametsuchi;
+
+/**
+ * @given pg_opt string with param1 and param2
+ * @when PostgresOptions object is created from given pg_opt string
+ * @then PostgresOptions object successfully created @and it contains param1 and
+ * param2
+ */
+TEST(PostgresOptionsTest, ParamExist) {
+  std::string pg_opt_string = "param1=val1 param2=val2";
+  auto pg_opt = PostgresOptions::create(pg_opt_string);
+  pg_opt.match(
+      [](iroha::expected::Value<PostgresOptions> options) {
+        auto param1 = options.value.getOption("param1");
+        ASSERT_TRUE(param1);
+        ASSERT_EQ(param1.value(), "val1");
+
+        auto param2 = options.value.getOption("param2");
+        ASSERT_TRUE(param2);
+        ASSERT_EQ(param2.value(), "val2");
+      },
+      [](auto) { FAIL() << "Creation of PostgresOptions object failed"; });
+}
+
+/**
+ * @given pg_opt string without non_existing_param
+ * @when PostgresOptions object is created from given pg_opt string
+ * @then PostgresOptions object successfully created @and doesn't contain
```

Comment:
The little issue about `@and`.
Better to separate and statements for readability:
```
@then PostgresOptions object successfully created
AND doesn't contain non_existing_param in it
```